integration/systemd/amazon-ecs-agent.service

```diff
@@ -0,0 +1,148 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the
+# "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and
+# limitations under the License.
+
+# A systemd unit file to run `ecs-agent` in a Docker container. This service
+# attempts to duplicate the functionality of the `ecs-init` Golang package. The
+# notable differences currently are:
+#
+# 1. The unit file downloads latest ECS Agent from
+# S3 at startup, if the files (ECS Agent tar / MD5 hash) are missing.
+# 2. The unit file does not currently handle the Agent's self-upgrade
+# functionality.
+# 3. The unit file does not clean up downloaded ECS Agents in /var/cache/ecs.
+
+[Unit]
+Description=Amazon ECS Agent
+Documentation=https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_agent.html
+Requires=docker.service
+After=docker.service
+
+[Install]
+WantedBy=multi-user.target
+
+[Service]
+# The logic around the ECS Agent restart:
+#   - Restart on failure
+#   - Wait 5 seconds between restart
+#   - If the ECS Agent restarts more then 10 times in 5 minutes,
+# it will stay failed
+Type=simple
+Restart=on-failure
+RestartSec=5s
+RestartPreventExitStatus=5
+StartLimitInterval=5min
+StartLimitBurst=10
+
+# It is recommended in production to version pin the ECS Agent 
+# to a known version vs running the latest flag on the docker container.
+# See https://github.com/aws/amazon-ecs-agent/releases for agent versions.
+
+# It is recommended to use a systemd drop-in file vs editing this line or file.
+# Your drop-in file could look like this:
+#
+#       [Service]
+#       Environment=ECS_AGENT_VERSION=v1.17.2
+#
+# Depending on the OS, it could be placed in 
+# /etc/systemd/system/amazon-ecs-agent.d , and named 
+# ecs_agent_version_override.conf
+Environment=ECS_AGENT_VERSION=latest
+Environment=DOWNLOAD_LOCATION=/var/cache/ecs
+
+ExecStartPre=-/usr/bin/echo "Amazon ECS Agent systemd unit file version 1.0.0"
+
+# Create the directories needed for the ECS Agent.
+ExecStartPre=/bin/mkdir -p /var/lib/ecs/dhclient /var/ecs-data /etc/ecs \
+    /var/cache/ecs
+
+# Test if /etc/ecs/ecs.config is there, if not create it.
+ExecStartPre=/bin/sh -c 'touch -a /etc/ecs/ecs.config'
+
+# Set the varable REGION for the ECS Agnet download from S3.
+ExecStartPre=/bin/sh -c "REGION=$(curl -s 169.254.169.254/latest/meta-data/placement/availability-zone | sed 's/.$//')"
+
+# Download ECS Agent from S3.
+ExecStartPre=/usr/bin/echo "Downloading the ECS Agent from S3, if missing"
+ExecStartPre=/bin/sh -c '\
+case $$REGION in \
```

Comment:
Does this actually work? REGION is set in a prior invocation of sh. That environment variable doesn't persist outside that shell.

Reply:
Fixed.   This lead to a few errors as well that have been fixed.